collector/src/main.rs

```diff
@@ -832,6 +942,79 @@ fn main_result() -> anyhow::Result<i32> {
             Ok(0)
         }
 
+        ("diff_local", Some(sub_m)) => {
+            // Mandatory arguments
+            let profiler = Profiler::from_name(sub_m.value_of("PROFILER").unwrap())?;
+            let rustc1 = sub_m.value_of("RUSTC1").unwrap();
+            let id1 = sub_m.value_of("ID1").unwrap();
+            let rustc2 = sub_m.value_of("RUSTC2").unwrap();
+            let id2 = sub_m.value_of("ID2").unwrap();
+
+            // Options
+            let build_kinds = build_kinds_from_arg(&sub_m.value_of("BUILDS"))?;
+            let cargo = sub_m.value_of("CARGO");
+            let exclude = sub_m.value_of("EXCLUDE");
+            let include = sub_m.value_of("INCLUDE");
+            let out_dir = PathBuf::from(sub_m.value_of_os("OUT_DIR").unwrap_or(default_out_dir));
+            let scenario_kinds = scenario_kinds_from_arg(sub_m.value_of("RUNS"))?;
+            let rustdoc = sub_m.value_of("RUSTDOC");
+
+            let mut toolchains = Vec::new();
+            for (id, rustc) in [(id1, rustc1), (id2, rustc2)] {
+                let (rustc, rustdoc, cargo) =
+                    get_local_toolchain(&build_kinds, rustc, rustdoc, cargo)?;
+                toolchains.push((id.to_owned(), rustc, rustdoc, cargo));
+            }
+
+            let benchmarks = get_benchmarks(&benchmark_dir, include, exclude)?;
+            let mut errors = BenchmarkErrors::new();
+            let mut remaining = benchmarks.len() * toolchains.len();
+            for (id, rustc, rustdoc, cargo) in &toolchains {
```

Comment:
Can we extract the inside of this loop into a function instead of duplicating it with profile local? It might merit a struct arg given the number of options likely we're passing in